packages/nimble-components/src/rich-text/editor/index.ts

```diff
@@ -335,6 +336,45 @@ export class RichTextEditor extends FoundationElement implements ErrorPattern {
 
     private createTiptapEditor(): Editor {
         const customLink = this.getCustomLinkExtension();
+        const validAbsoluteLinkRegex = /^https?:\/\//i;
+
+        /**
+         * @param slice contains the Fragment of the copied content. If the content is a link, the slice contains anchor text node with link mark.
+         * ProseMirror reference for `transformPasted`: https://prosemirror.net/docs/ref/#view.EditorProps.transformPasted
+         */
+        const transformPasted = (slice: Slice): Slice => {
```

Comment:
nit: my personal preference would be for `transformPasted` and `updateLinkNodes` to be `private` methods on the class rather than arrow functions. I believe that should be a trivial change (I don't see any state that they're capturing) but if there's some reason that the current approach is necessary or preferable then I don't feel too strongly.

Reply:
We were able to move the `updateLinkNodes` to be a private method but moving `transformPasted`, creates an issue. When we tried moving this to a private method, the function was not added to the [`transformPasted`](https://prosemirror.net/docs/ref/#view.EditorProps.transformPasted) of `editorProps` and renders the hyperlink as such and not transformed to an absolute link.